osquery/config/parsers/file_paths.cpp

```diff
@@ -81,20 +85,48 @@ Status FilePathsConfigParserPlugin::update(const std::string& source,
     data_.add("file_accesses", arr);
   }
 
-  // We know this top-level is an Object.
-  const auto& file_paths = config.at("file_paths").doc();
-  if (file_paths.IsObject()) {
-    for (const auto& category : file_paths.GetObject()) {
-      if (category.value.IsArray()) {
-        for (const auto& path : category.value.GetArray()) {
-          std::string pattern = path.GetString();
-          if (pattern.empty()) {
-            continue;
+  if (config.count("file_paths") > 0) {
+    // We know this top-level is an Object.
+    const auto& file_paths = config.at("file_paths").doc();
+    if (file_paths.IsObject()) {
+      for (const auto& category : file_paths.GetObject()) {
+        if (category.value.IsArray()) {
+          for (const auto& path : category.value.GetArray()) {
+            std::string pattern = path.GetString();
+            if (pattern.empty()) {
+              continue;
+            }
+
+            std::string name = category.name.GetString();
+            replaceGlobWildcards(pattern);
+            Config::get().addFile(source, name, pattern);
           }
+        }
+      }
+    }
+  }
 
-          std::string name = category.name.GetString();
-          replaceGlobWildcards(pattern);
-          Config::get().addFile(source, name, pattern);
+  if (config.count("file_paths_query") > 0) {
+    // We know this top-level is an Object.
+    const auto& path_query_node = config.at("file_paths_query").doc();
+    if (path_query_node.IsObject()) {
+      for (const auto& category : path_query_node.GetObject()) {
+        if (category.value.IsArray()) {
+          for (const auto& query : category.value.GetArray()) {
+            auto sql = SQL(query.GetString());
+            if (!sql.ok()) {
+              LOG(ERROR) << "Could not add file_paths using file_paths_query '"
+                         << query.GetString()
+                         << "': " << sql.getMessageString();
+            } else {
+              for (auto& row : sql.rows()) {
+                std::string name = category.name.GetString();
+                std::string path = row["path"];
```

Comment:
Should we consider case when key `"path"` is absent in the query result for some reason?
If so, `path` will be an empty string, which is probably not good.